### PR TITLE
[Longformer] Better handling of global attention mask vs local attention mask

### PR DESCRIPTION
**Pull request by patrickvonplaten**
This PR extends Longformer's API to also take a `global_attention_mask` besides the usual `attention_mask` as an input as discussed with @thomwolf @ibeltagy. 

Docs are updated.

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Regarding the Multiple Choice global attention mask:

> @patil-suraj, we can leave it to the user, or we can just do as you suggested earlier, put global attention on the question and all choices, which should work.
> 
> @patrickvonplaten, what do you think?

Regarding the multiple choice, I think we usually have the following tensor:
```
[
[ context, choice_a],
[ context, choice_b],
[ context, choice_c], 
....
] 
```

see here: https://github.com/huggingface/transformers/blob/9c17256447b91cf8483c856cb15e95ed30ace538/examples/multiple-choice/utils_multiple_choice.py#L529

So I'd suggest if no `global_attention_mask` is provided by the user, we initialize the `global_attention_mask` so that all choice contexts do global attention. If the user wants a different global attention he has now the possibility to define it himself.

**Comment by patil-suraj** [comment]
> Ok, checked the notebook: https://colab.research.google.com/drive/1m7eTGlPmLRgoPkkA7rkhQdZ9ydpmsdLE?usp=sharing and results are the same as before so no breaking changes.
> 
> Good to merge for me!

This is great !

Just noticed one typo in the first line
> This notebook shows how `nlp` can be leveraged `nlp` to evaluate Longformer on TriviaQA

File: src/transformers/modeling_longformer.py [hunk 1]
```diff
@@ -1132,8 +1169,8 @@ def forward(
         from transformers import LongformerTokenizer, LongformerForTokenClassification
         import torch
 
-        tokenizer = LongformerTokenizer.from_pretrained('longformer-base-4096')
-        model = LongformerForMultipleChoice.from_pretrained('longformer-base-4096')
+        tokenizer = LongformerTokenizer.from_pretrained('allenai/longformer-base-4096')
+        model = LongformerForMultipleChoice.from_pretrained('allenai/longformer-base-4096')
         choices = ["Hello, my dog is cute", "Hello, my cat is amazing"]
```
Comment (ibeltagy) [inline src/transformers/modeling_longformer.py:1174]: maybe use an example that has `context` in addition to the `choices` with a comment showing where the automatic global attention is going to be.

**Review by ibeltagy — COMMENTED** [review]
Looks great. I left a couple of small comments. Thanks, Patrick.

**Review by thomwolf — APPROVED** [review]

File: src/transformers/modeling_longformer.py (hunk 1 above)
Comment (patrickvonplaten) [reply src/transformers/modeling_longformer.py]: very good point!
